app/controllers/events_controller.rb

```diff
@@ -0,0 +1,63 @@
+class EventsController < ApplicationController
+  before_action :require_login!, only: %i[new create]
+  before_action :set_event, only: %i[show destroy edit update]
+  before_action :owner?, only: %i[destroy edit]
+
+  def index
+    @events = Event.all
+  end
```

Comment:
* [ ] list past events, i.e. `@past_events = Event.previous_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [ ] list ucomping events, i.e. `@upcoming_events = Event.upcoming_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

**_don't add `upcoming_events` and `previous_events` in the controller, these should be in the events model._**

Reply:
Thank you